File: benchmarks/oopsla25/subjects/042.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f042(const char *s) {
    size_t n = strlen(s);
    if (n > 2) {
        fprintf(stderr, "Exception\n");
        exit(1);
    } else {
        char c1 = s[n - 1];
        char c2 = s[n - 2];
        return;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f042(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/042.py
import sys

def f042(s: str):
  n = len(s)
  if n > 2:
    raise Exception
  else:
    c1 = s[n-1]
    c2 = s[n-2]
    return

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f042(my_string))
